src/python/WMCore/ReqMgr/Utils/Validation.py

```diff
@@ -186,3 +187,35 @@ def get_request_template_from_type(request_type, loc="WMSpec.StdSpecs"):
 
     result = create_json_template_spec(specArgs)
     return result
+
+def validateOutputDatasets(outDsets, dbsUrl):
+    """
+    Validate output datasets after all the other arguments have been
+    locally update during assignment
+    """
+    datatier = []
+    for dataset in outDsets:
+        tokens = dataset.split("/")
+        procds = tokens[2]
+        datatier.append(tokens[3])
+        try:
+            procdataset(procds)
+        except AssertionError as ex:
+            msg = "Bad output dataset name, check the processed dataset.\n %s" % str(ex)
```

Comment:
It seems there is a case procdataset returns False ("" or None)

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Lexicon.py#L199

Shoudn't we catch that case as well?